onnxruntime/core/providers/cpu/nn/string_normalizer.cc

```diff
@@ -201,7 +201,16 @@ class Utf8Converter {
 
 #endif
 
-const std::string default_locale("en_US.UTF-8");  // All non-MS
+#if defined(__APPLE__)
+#include <TargetConditionals.h>
+#if TARGET_OS_IPHONE || TARGET_IPHONE_SIMULATOR
```

Comment:
in TargetConditionals.h, it says that `TARGET_IPHONE_SIMULATOR` is deprecated and the same as `TARGET_OS_SIMULATOR`.
```suggestion
#if TARGET_OS_IPHONE || TARGET_OS_SIMULATOR
```